backend-shared/db/clusteraccess.go

```diff
@@ -117,8 +117,8 @@ func (dbq *PostgreSQLDatabaseQueries) DeleteClusterAccessById(ctx context.Contex
 
 func (dbq *PostgreSQLDatabaseQueries) ListClusterAccessesByManagedEnvironmentID(ctx context.Context, managedEnvironmentID string, clusterAccesses *[]ClusterAccess) error {
 
-	if err := validateQueryParamsEntity(clusterAccesses, dbq); err != nil {
-		return err
+	if clusterAccesses == nil {
```

Comment:
What's the reason for changing this?  Will there be a situation when dbq.dbConnection = nil ?

Reply:
Thanks. I've reverted the change. I didn't realize we were checking the internal connection field `.dbConnection`. But the original reason I didn't want to use `validateQueryParamsEntity` is because we are converting the clusterAccesses a slice to an interface when we pass it as an argument. Even though the clusterAccesses can be `nil`, when passed to the `validateQueryParamsEntity` the interface will no longer be nil and the validation fails. Because in Go, an interface can be considered as a pair of <type,value> and an interface is `nil` only when both its type and value are `nil`. It doesn't affect the test coverage but we might need to update the function to check if the interface value is nill.

```

func (dbq *PostgreSQLDatabaseQueries) ListClusterAccessesByManagedEnvironmentID(ctx context.Context, managedEnvironmentID string, clusterAccesses *[]ClusterAccess) error {

        // clusterAccesses can be nil here. But in the below function, it is used as an interface 
        // and only the interface value will be nil and interface type still says that is a slice. 
	if err := validateQueryParamsEntity(clusterAccesses, dbq); err != nil {
		return err
	}

```